test/e2e/Makefile

```diff
@@ -0,0 +1,82 @@
+# Copyright 2020 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# If you update this file, please follow:
+# https://suva.sh/posts/well-documented-makefiles/
+
+# Use GOPROXY environment variable if set
+
+.DEFAULT_GOAL:=help
+
+GOPROXY := $(shell go env GOPROXY)
+ifeq ($(GOPROXY),)
+GOPROXY := https://proxy.golang.org
+endif
+export GOPROXY
+
+REPO_ROOT := $(shell git rev-parse --show-toplevel)
+
+help:  ## Display this help
+	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make \033[36m<target>\033[0m\n"} /^[a-zA-Z_-]+:.*?##/ { printf "  \033[36m%-15s\033[0m %s\n", $$1, $$2 } /^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } ' $(MAKEFILE_LIST)
+
+## --------------------------------------
+## Binaries
+## --------------------------------------
+
+TOOLS_DIR := $(REPO_ROOT)/hack/tools
+BIN_DIR := bin
+TOOLS_BIN_DIR := $(TOOLS_DIR)/$(BIN_DIR)
+GINKGO := $(TOOLS_BIN_DIR)/ginkgo
+
+.PHONY: ginkgo
+ginkgo:
+	cd $(TOOLS_DIR) && go build -tags=tools -o $(BIN_DIR)/ginkgo github.com/onsi/ginkgo/ginkgo
+
+## --------------------------------------
+## Testing
+## --------------------------------------
+
+TEST_E2E_DIR := $(REPO_ROOT)/test/e2e/
+
+GINKGO_FOCUS  ?=
+GINKGO_NODES  ?= 1
+SKIP_RESOURCE_CLEANUP ?= false
+GINKGO_NOCOLOR ?= false
+NODE_IMAGE ?= kindest/node:v1.16.15
+PROVIDER ?=
+CHART_PATH ?= $(REPO_ROOT)/charts/secrets-store-csi-driver
+MANIFESTS_DIR ?= $(REPO_ROOT)/test/e2e/data
+KINDCONFIG_PATH ?= $(MANIFESTS_DIR)/kind-config.yaml
+ifeq ($(PROVIDER),azure)
+TRY_LOAD_IMAGES ?= \
+	mcr.microsoft.com/k8s/csi/secrets-store/provider-azure:0.0.8 \
+	nginx
+else ($(PROVIDER),vault)
+TRY_LOAD_IMAGES ?= \
```

Comment:
images in try_load is just trying to load local pulled images to kind clusters.  otherwise, image will be pulled when resources is deployed in kubernetes. the former makes running e2e faster.